adapters/oidc/js/src/main/resources/keycloak.js

```diff
@@ -313,6 +374,14 @@
                 url += '&kc_locale=' + encodeURIComponent(options.kcLocale);
             }
 
+            if (options && kc.pkceMethod) {
+                var codeVerifier = generateCodeVerifier(96);
+                localStorage.setItem('kc-pkceCodeVerifier', codeVerifier);
```

Comment:
@mposolda correct me if I'm wrong. But wouldn't be better if we use cookie storage or maybe session storage here?

Reply:
`sessionStorage` would be sufficient and also automatically cleared when the browser window is closed.
A "client-side" session cookie, that doesn't get sent to the server might also be useful.

Reply:
This should leverage the callbackStorage and add the details to the callback like we do for state param, not add additional/separate info.

Reply:
localStorage isn't always available. That is what we used for the callback initially, but then we added this thing https://github.com/keycloak/keycloak/blob/master/adapters/oidc/js/src/main/resources/keycloak.js#L1537 which uses localstorage if avail, or cookie storage if not. I would just use the callbackStorage, or add a codeVerifierStorage that uses the same  createCallbackStorage.

Reply:
Actually, why not just add it to the callbackState?

Reply:
+1 for add to the callbackState as that's already used